client/src/modules/blockEditor/actions.ts

```diff
@@ -114,11 +174,51 @@ export const removeExportOfSelection = () => {
         return;
       }
 
-      const change = block.value
-        .change()
-        .unwrapInlineByKey(matchingNodes[0].key, {
-          data: { pointerId: hoveredItem.id }
-        });
+      const nodeToRemove = matchingNodes[0];
+
+      let change;
+      change = block.value.change();
+
+      // delete spacer at start
```

Comment:
I think I already asked this somewhere (in that case, just point me to your previous response): Why do we need to manually delete the spacers below? Why can't we just remove the export and then rely on normalization to fix the spacing?

Reply:
Just because normalization doesn't remove characters. We could change it so that it does, but then users couldn't insert thin spaces (or whatever the spacer character is) in other positions.

Reply:
Is there some way to attach metadata to the spacer characters so that it's distinguished from normal ones?

Having normalization take care of the entire export space formatting seems like a substantial simplification to me.

Reply:
With Slate, as far as I can figure out, it's actually really complicated to add metadata like that. Basically, you can add a mark, which is like formatting, so it is similar to bold or italics. In the case of metadata, the mark just wouldn't correspond to any visual difference. 

But then you have to add a lot of logic to make sure new text you insert doesn't pick up the mark (think of how you can put your cursor next to bold text, start typing, and insert more bold text).

I would be OK with just saying that users can't use the spacer character, especially when the spacer character is "non-breaking thin space". I could also put a comment in the file that determines the spacer character, warning developers that the spacer character is not something the user can manipulate (so don't set it to normal space, for example). If we did this, then we could expand the normalization process in the way you're thinking.

Reply:
Sounds good. To be extra-safe, we could programmatically prevent users from inserting the spacer character. That's probably overkill for now, but you can make an issue for it.

Reply:
https://github.com/oughtinc/mosaic/pull/145/commits/01608fdbc36018c097e3f22d9c7c172939c703fc